### PR TITLE
Prepare provider for v0.1.0 secrets store

### PR DESCRIPTION
**Pull request by vaegt**
As newer releases of the secrets-store won't allow anything but gRPC providers, this PR is the implementation of the spring cloud config provider with gRPC.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,110 +1,30 @@
 package main
 
 import (
-	"encoding/json"
-	"io"
-	"io/ioutil"
-	"os"
-
+	"fmt"
+	"github.com/freenowtech/secrets-store-csi-driver-provider-spring-cloud-config/pkg/provider"
```
Comment (wndhydrnt) [inline main.go:5]: Please apply `goimports` to this file.

File: pkg/provider/server.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	log "github.com/sirupsen/logrus"
```
Comment (wndhydrnt) [inline pkg/provider/server.go:7]: Please apply `goimports` to this file.

File: pkg/provider/server.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	}
+	defer file.Close()
+
+	_, err = io.Copy(file, content)
+	if err != nil {
+		return nil, fmt.Errorf("secrets store csi driver failed to mount %s at %s: %w", fileName, req.GetTargetPath(), err)
+	}
+	log.Infof("secrets store csi driver mounted %s", fileName)
+	log.Infof("mount point: %s", req.GetTargetPath())
+	//out.Files = append(out.Files, &v1alpha1.File{
```
Comment (wndhydrnt) [inline pkg/provider/server.go:168]: Is this commented code block still needed? I can see that `out.Files` is being set in line 154. So maybe it is not needed.

File: pkg/provider/server.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	}
+
+	if a.fileType == "" {
+		return fmt.Errorf("fileType is not set")
+	}
+
+	return nil
+}
+
+// NewMocKCSIProviderServer returns a mock csi-provider grpc server
```
Comment (wndhydrnt) [inline pkg/provider/server.go:55]: Please adjust the doc string to match the function.

File: pkg/provider/server.go [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+func (m *SpringCloudConfigCSIProviderServer) SetObjects(objects map[string]string) {
+	var ov []*v1alpha1.ObjectVersion
+	for k, v := range objects {
+		ov = append(ov, &v1alpha1.ObjectVersion{Id: k, Version: v})
+	}
+	m.objects = ov
+}
+
+// SetFiles sets provider files to return on Mount
+func (m *SpringCloudConfigCSIProviderServer) SetFiles(files []*v1alpha1.File) {
```
Comment (wndhydrnt) [inline pkg/provider/server.go:83]: I am a bit skeptical about the usefulness of this function. What catches my eye is that it modifies `m.files`. Given that this code runs as a server and requests can come in in any order, this opens the door for requests overriding each others state.
Also, the interface [CSIDriverProviderServer](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/v0.1.0/provider/v1alpha1/service.pb.go#L756-L763) does not describe this function.
Does code of the provider call this function anywhere?

**Review by wndhydrnt — COMMENTED** [review]
Found one potential race condition. Also it is unclear to me if the code is able to perform its function the way it is implemented right now.

Find some formatting issues too.

File: pkg/provider/server.go [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+		return nil, fmt.Errorf("failed to retrieve secrets for %s: %w", fileName, err)
+	}
+	defer content.Close()
+
+	out := &v1alpha1.MountResponse{
+		ObjectVersion: m.objects,
+		Error: &v1alpha1.Error{
+			Code: m.errorCode,
+		},
+		Files: m.files,
```
Comment (wndhydrnt) [inline pkg/provider/server.go:123]: I'm having trouble understanding the use of `m.files`. It looks to me like the slice `m.files` will always be empty. Is there a need to set `v1alpha1.MountResponse.Files`?

File: pkg/provider/server.go [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+	fileName := fmt.Sprintf("%s-%s.%s", attrib.Application, attrib.Profile, attrib.FileType)
+	content, err := m.springCloudConfigClient.GetConfig(attrib)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve secrets for %s: %w", fileName, err)
+	}
+	defer content.Close()
+
+	out := &v1alpha1.MountResponse{
+		ObjectVersion: m.objects,
```
Comment (wndhydrnt) [inline pkg/provider/server.go:119]: Similar to my comment regarding `m.files`, it looks like the slice `m.objects` is always empty. Does it make sense to use it here?

**Review by wndhydrnt — COMMENTED** [review]
It looks like some slices are always empty. I currently cannot judge if this is a problem or not, but its something to clarify.

File: pkg/provider/server.go (hunk 6 above)
Comment (vaegt) [reply pkg/provider/server.go:123]: `m.files` and `m.objects` are only necessary if we want a correct secret syncing implementation, but as we are currently not in need for this, I will remove them

**Review by wndhydrnt — APPROVED** [review]

**Pull request merged**